### DISCUSSION TITLE
[New Feature] Unit type grouping for type select

### DISCUSSION THREAD

**Discussion by Rampastring** [Vinifera-Developers/Vinifera#1385] [Feature Requests]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I have attached as much information as possible *(screenshots, videos, links, etc)*.

### Description

Some mods have units that are practically identical in functionality with other units. For example, DTA has a Minigunner unit for GDI and Nod and a Rifle Infantry unit for the Allies and Soviets, with their only difference being different voice lines and different weapon firing sounds.

It'd be preferable if these kinds of units could all be selected at the same time when you press the 'Type select' hotkey on one of them. For example, if I select a GDI Minigunner and press T (type select hotkey) while there's 5 Soviet Rifle Infantry on the screen, all of the units controlled by me, it'd be nice if the game selected all of them as they're identical in functionality.

The modder should be free to define which units are grouped with each other.

### Proposed Code (optional)

One idea that comes to mind is linking them at a `TechnoTypeClass` level. A `TechnoType` could have a list of techno types linked with it, like this

```ini
[E1]
LinkedObjects=E1N,E1A,E1S ; link with basic infantry of other factions for type selection
```

This could, however, lead to duplicate code, as `E1N`, `E1A` and `E1S` would also need the same linking code added to them (unless a bi-directional link would be established automatically by Vinifera).

Another option I think could be introducing a bunch of new keys for linked unit groups

```ini
[General]
LinkedUnitGroup0=E1,E1N,E1A,E1S
LinkedUnitGroup1=E2,E2S
LinkedUnitGroup2=E3,E3N,E3A,E3S
... and so on
```

**Comment by Onidotmoe** [comment]
I think a better key name would be "Similar"

or even better would be defining the units as having a common "base" unit and deriving from that for selection and other purposes.

Like this hierarchy :
Object -> Unit -> Infantry -> Light Infantry -> Soldier -> GDI

So when you want to make a new Soldier variant you'd have it derive from the Soldier Class instead of from E1

**Comment by Onidotmoe** [comment]
Kinda mentioned in #860 but using this key with a "." :

`Selection.Similar=Unit1,Unit2,Unit2`

 in the middle would probably be best practice, keeping the key by the unit entry would be best as it allows multiple mods to change the behavior without overriding the global value, and also keeping the relevant key by the unit/building would just make it easier to work with.
